### ISSUE TITLE
UnsatisfiedLinkError for < API 24 Devices

### ISSUE THREAD

**Issue by mralbean** [google-ar/arcore-android-sdk#111]
In the ARCore reference docs, there is an exception, [UnavailableAndroidVersionNotSupportedException](https://developers.google.com/ar/reference/java/com/google/ar/core/exceptions/UnavailableAndroidVersionNotSupportedException), which is described to be thrown when you try to run ARCore on a device under API level 24. When I run the sample applications on an API 23 device though, I get the following crash:

```
E  FATAL EXCEPTION: main
E  Process: com.google.ar.core.examples.java.computervision, PID: 5534
E  java.lang.UnsatisfiedLinkError: dlopen failed: library "libcamera2ndk.so" not found
E      at java.lang.Runtime.loadLibrary(Runtime.java:372)
E      at java.lang.System.loadLibrary(System.java:1076)
E      at com.google.ar.core.Session.<init>(Session.java:84)
E      at com.google.ar.core.examples.java.computervision.MainActivity.onCreate(MainActivity.java:125)
E      at android.app.Activity.performCreate(Activity.java:6251)
E      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
E      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
E      at android.app.ActivityThread.-wrap11(ActivityThread.java)
E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
E      at android.os.Handler.dispatchMessage(Handler.java:102)
E      at android.os.Looper.loop(Looper.java:148)
E      at android.app.ActivityThread.main(ActivityThread.java:5417)
E      at java.lang.reflect.Method.invoke(Native Method)
E      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

What’s the purpose of UnavailableAndroidVersionNotSupportedException if it is not thrown when running on a device under the minimum supported version? Should we instead be checking the API level ourselves before attempting to create an ARCore session?

**Comment by inio** [comment]
Thanks for the report!  This sounds like a legit bug.

**Comment by inio** [comment]
This should be fixed with the 1.0 release today.  Additionally, we've [introduced an API for checking device compatibility](https://devsite.googleplex.com/ar/develop/java/enable-arcore#check_supported) which any app targeting SDKs earlier than 24 must use (ARCore-Required apps must have midSdk ≥ 24).
